app/models/miq_ae_domain.rb

```diff
@@ -27,6 +27,11 @@ class MiqAeDomain < MiqAeNamespace
 
   include TenancyMixin
 
+  def self.with_tenant(tenant_id)
+    tenant = Tenant.find(tenant_id)
```

Comment:
What if ```tenant_id``` does not exist?

Reply:
well in that case it makes no sense to call this method w/o a valid tenant

the code would raise an exception now

@lfu: do you think that returning an empty `ActiveRecord::Relation` is a better solution?

Reply:
@lfu @martinpovolny I think to throw exception is what we need here - because `tenant_id == true` is (as you mentioned ) something what is unexpected.

Reply:
Sounds good to me.

Reply:
@lfu Looks good, but should this be `with_tenants` since it potentially returns multiple tenants?

Reply:
Ah, I think it is ```with_tenant``` with a single tenant id.

Reply:
Got it, thanks @lfu.